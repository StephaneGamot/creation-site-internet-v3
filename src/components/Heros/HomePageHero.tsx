"use client"

import { useTranslations } from 'next-intl';
import { motion } from 'framer-motion'
import KnowMoreButton from '../Buttons/KnowMoreButton'

export default function HomePageHero() {
  const t = useTranslations('home');

  return (
    <section role="region" aria-labelledby="home-hero-title" className="text-center pt-12 max-w-4xl mx-auto mb-12">
    <motion.h1
     id="home-hero-title"
      initial={{ opacity: 0, y: 40 }}
      animate={{ opacity: 1, y: 0 }}
      transition={{ duration: 0.8 }}
      className="text-4xl md:text-5xl font-title font-semibold mb-8"
    >
{t('heroTitle')}
    </motion.h1>
    <p role="region" className="text-lg md:text-xl text-gray-700 mb-10">
    {t('heroSubtitle')}
    </p>
  <KnowMoreButton />
  </section>
  )
}
