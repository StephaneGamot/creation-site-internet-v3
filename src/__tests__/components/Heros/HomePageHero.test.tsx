// src/__tests__/components/HomePageHero.test.tsx
import React, { forwardRef } from "react";
import { render, screen } from "@testing-library/react";
import HomePageHero from "@/components/Heros/HomePageHero";
import renderer from "react-test-renderer";
import { axe } from "jest-axe";

// ✅ Mock de next-intl (t)
jest.mock("next-intl", () => ({
  useTranslations: () => {
    const t = (key: string) => {
      if (key === "heroTitle") return "Création de site web impactant";
      if (key === "heroSubtitle") return "Sites modernes, élégants et optimisés SEO.";
      return "";
    };
    return t;
  },
  useLocale: () => "fr",
}));

// ✅ Mock de framer-motion
jest.mock("framer-motion", () => ({
  motion: {
    h1: forwardRef<HTMLHeadingElement, React.HTMLProps<HTMLHeadingElement>>((props, ref) => (
      <h1 ref={ref} {...props} />
    )),
  },
}));

// ✅ Mock du bouton KnowMoreButton
jest.mock("@/components/Buttons/KnowMoreButton", () => () => (
  <a href="/fr/contact">En savoir plus</a>
));


describe("HomePageHero", () => {
  it("affiche le titre, le sous-titre et le bouton", () => {
    render(<HomePageHero />);
    expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent("Création de site web impactant");
    expect(screen.getByText("Sites modernes, élégants et optimisés SEO.")).toBeInTheDocument();
    expect(screen.getByText("En savoir plus")).toBeInTheDocument();
  });

  it("passe les tests d’accessibilité", async () => {
    const { container } = render(<HomePageHero />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it("correspond au snapshot", () => {
    const tree = renderer.create(<HomePageHero />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
