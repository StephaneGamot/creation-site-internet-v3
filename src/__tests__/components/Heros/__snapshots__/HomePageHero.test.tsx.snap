// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`HomePageHero correspond au snapshot 1`] = `null`;
